client/src/helpers/APICalls/bookingService.ts

```diff
@@ -0,0 +1,74 @@
+import { BookingApiData, BookingRequest } from '../../interface/BookingApiData';
+import { FetchOptions } from '../../interface/FetchOptions';
+import { Modify } from './../../context/useRequestContext';
+
+interface UpdateBooking {
+  accepted?: boolean;
+  declined?: boolean;
+}
+
+export const getBookings = async (): Promise<Array<BookingRequest>> => {
+  const fetchOptions: FetchOptions = {
+    method: 'GET',
+    credentials: 'include',
+  };
+  const res = await fetch('/request', fetchOptions);
+  if (res.status !== 200) {
+    handleError(res);
+    throw new Error();
+  }
+  return await res.json();
+};
+
+export const updateBooking = async (value: Modify, _id: string): Promise<BookingRequest> => {
+  const body: UpdateBooking = {};
+
+  if (value === 'Accept') body.accepted = true;
+  if (value === 'Decline') body.declined = true;
+
+  const fetchOptions: FetchOptions = {
+    method: 'PUT',
+    headers: { 'Content-Type': 'application/json' },
+    credentials: 'include',
+    body: JSON.stringify(body),
+  };
+  const res = await fetch(`/request/${_id}`, fetchOptions);
+  if (res.status !== 200) {
+    handleError(res);
+    throw new Error();
+  }
+  return await res.json();
+};
```

Comment:
The `PUT` verb would be misleading here as we only update a portion of the resource and not replace it completely.

Reply:
Thanks for the heads up. I'll make the necessary adjustment